### DISCUSSION TITLE
blog/2023/04/04/make-postgres-an-application-server-gamified/

### DISCUSSION THREAD

**Discussion by giscus** [yrashk/yrashk.github.io#1] [General]
# blog/2023/04/04/make-postgres-an-application-server-gamified/

Writings on software development and work practices, entrepreneurship, tech in general.

https://yrashk.com/blog/2023/04/04/make-postgres-an-application-server-gamified/

**Comment by ggaughan** [comment]
https://gitlab.com/-/snippets/2524619

**Reply by yrashk** [reply]
@ggaughan congratulations for claiming Challenge `#3`! I will be reaching out shortly :)

**Comment by ggaughan** [comment]
https://gitlab.com/-/snippets/2524682

**Reply by yrashk** [reply]
@ggaughan Great job! With a small nitpick of losing the plain text format response if html is not expected (that's an easy fix), Challenge `#1` is yours, too.

**Comment by kartikynwa** [comment]
Authorization is very rudimentary. Revisiting plpgsql broke my brain a little bit.

https://gist.github.com/kartikynwa/d985d988ad8f37c1c724176ccc5c87c6

**Reply by yrashk** [reply]
Great, congratulations!

On the plpgsql end, I believe that you don't actually have to use it to achieve this. I

I would recommend looking at cascading queries primitive (used at the bottom of the MOTD source code) and [documented here](https://docs.omnigres.org/omni_httpd/intro/#getting-started) as pure SQL could have been used to check if there's a valid authorization. It can be more efficient as it won't have to be interpreted by the plpgsql extension but instead inlined into the query.

Anyone wants to submit their solution based on this idea?

That being said, plpgsql one is definitely more familiar to those comfortable with the imperative approach.

**Reply by ebraminio** [reply]
Late to the party, here is my SQL only version
https://gist.github.com/ebraminio/29949131c691e6abd76bf21a9f512f23 😊

**Reply by yrashk** [reply]
@ebraminio, that's a a good way to solve this with SQL!
